azurelinuxagent/agent.py

```diff
@@ -62,11 +65,25 @@ def __init__(self, verbose, conf_file_path=None):
         #Init log
         verbose = verbose or conf.get_logs_verbose()
         level = logger.LogLevel.VERBOSE if verbose else logger.LogLevel.INFO
+        log_dir = self.osutil.get_agent_log_dir()
+        try:
+            if os.path.isfile(log_dir):
+                raise Exception("{0} is a file".format(log_dir))
+            if not os.path.isdir(log_dir):
+                fileutil.mkdir(log_dir, mode=0o755, owner="root")
+        except Exception as e:
+            logger.error(
+                "Exception occurred while creating "
+                "log directory {0}: {1}".format(log_dir, e))
+        log_file = os.path.join(log_dir, WAAGENT_LOG_FILE)
```

Comment:
In the scenario that `log_dir` is actually a file (i.e. `os.path.isfile(log_dir) == true`), the `os.path.join` would just append the log filename to the log_dir file. That inturn would throw an error later when trying to write to file. 

Eg: 
```python
with open('/home/larohra/test/1.log', 'w') as fh:
...     fh.write("test123")
...     
Traceback (most recent call last):
  File "<input>", line 1, in <module>
IOError: [Errno 20] Not a directory: '/home/larohra/test.txt/1.log'
```

We should handle this scenario too. If incase log_dir is a file, maybe just default to our `/var/log` directory?

Reply:
But this is under the assumption that `log_dir` is a file, and someone did not understand the word `dir`. It is supposed to be a directory :)

I didn't understand your concern here - you fear that the OSUtil will have file, in place of a directory?

Reply:
We have a check above - 
```python
if os.path.isfile(log_dir):
                raise Exception("{0} is a file".format(log_dir)
```

And this exception is just being eaten and logging an error and just moving on. If this scenario is actually true, then the log_appender initialization would throw an error which (haven't checked) could bring down the agent. That is my main concern. Its a far fetched case but I feel we sjpi;d handle it.

Reply:
>I didn't understand your concern here - you fear that the OSUtil will have file, in place of a directory?

Yeah basically